server/src/Infrastructure/Persistence/Repositories/Repository.cs

```diff
@@ -12,98 +14,70 @@ public abstract class Repository<TEntity> : IRepository<TEntity> where TEntity :
     public Action<TEntity> OnInstanceAdded = delegate { };
     public Action<TEntity> OnInstanceUpdated = delegate { };
 
-    private readonly string _filePath;
-    private readonly IFileService _fileService;
-    private readonly ApplicationDbContext _dbContext;
+    protected readonly ApplicationDbContext _dbContext;
 
-    protected Repository(string fileName, IFileService fileService, ApplicationDbContext dbContext)
+    protected Repository(ApplicationDbContext dbContext)
     {
-        _fileService = fileService;
-        _filePath = Path.Combine(
-            Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData),
-            "StudyZen",
-            $"{fileName}.json");
         _dbContext = dbContext;
 
         OnInstanceAdded += AuditableEntityInterceptor.SetCreateStamp;
         OnInstanceUpdated += AuditableEntityInterceptor.SetUpdateStamp;
     }
 
-    public void Add(TEntity instance)
+    public async Task Add(TEntity instance)
     {
-        var entitySet = GetEntitySet();
-
-        instance.Id = ++entitySet.TotalCount;
-        OnInstanceAdded(instance);
+        try
+        {
+            OnInstanceAdded(instance);
 
-        Log.Information("Adding {0} with id {1}", typeof(TEntity).Name, instance.Id);
-        entitySet.Instances.Add(instance);
+            await _dbContext.Set<TEntity>().AddAsync(instance);
```

Comment:
I think just a regular Add can be used, since it does not use database but only marks the entity state as added.